src/modm/math/utils/integer_traits.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <limits>
+#include <type_traits>
+
+#include <cmath>
+namespace modm
+{
+	template <int Bits>
+	struct uint_t
+	{
+		using least = std::conditional_t<
+			(Bits <= 8),uint8_t, std::conditional_t<
+			(Bits <= 16), uint16_t, std::conditional_t<
+			(Bits <= 32), uint32_t, std::enable_if_t<
+			(Bits <= 64), uint64_t>>>>;
+	};
```

Comment:
1. Should be unindented in namespace.
2. Should this be in the `modm::detail` namespace?

Reply:
> 2. Should this be in the modm::detail namespace?
STL traits usually offer both, the **type** and the **using shorthand**. I think that's fine but i've putted `most_digits()` into `namespace detail` ;)